bin/extractors/dratf_extractor.py

```diff
@@ -177,17 +198,27 @@ def extract_all_elements(self, extracted_yang_models: list):
                 extract_elem(module_fname, self.draft_elements_path, 'grouping')
                 extract_elem(module_fname, self.draft_elements_path, 'identity')
 
-    def dump_incorrect_drafts(self, public_directory: str):
-        """ Dump names of the IETF drafts with xym extraction error to
-        problematic_drafts.json file.
-        """
-        base_path = os.path.join(public_directory, 'drafts')
-        file_path = os.path.join(base_path, 'problematic_drafts.json')
+    def dump_incorrect_drafts(self, public_directory: str, send_emails_about_problematic_drafts: bool = True):
+        """Dump names of the IETF drafts with xym extraction error to problematic_drafts.json file."""
+        drafts_directory = os.path.join(public_directory, 'drafts')
+        os.makedirs(drafts_directory, exist_ok=True)
+        file_path = os.path.join(drafts_directory, 'problematic_drafts.json')
+        if send_emails_about_problematic_drafts:
+            if not os.path.exists(file_path):
+                with open(file_path, 'w') as f:
+                    f.write('{}')
+                old_incorrect_drafts_keys = []
+            else:
+                with open(file_path, 'r') as f:
+                    old_incorrect_drafts_keys = json.load(f).keys()
+            self._send_email_about_new_problematic_drafts(old_incorrect_drafts_keys)
+        with open(file_path, 'w') as writer:
+            json.dump(self.drafts_missing_code_section, writer)
```

Comment:
This looks like it would only keep the error from the latest run. We want to keep the errors from previous runs as well.

Reply:
Actually never mind, all the errors should get generated on every run anyway.